src/code/z_parameter.c

```diff
@@ -1145,26 +1149,474 @@ void Inventory_ChangeAmmo(s16 item, s16 ammoChange) {
+            if (!(gSaveContext.save.weekEventReg[14] & 8)) {
+                gSaveContext.save.playerData.magic -= 2;
+            }
+            if (gSaveContext.save.playerData.magic <= 0) {
+                gSaveContext.save.playerData.magic = 0;
+            }
+            gSaveContext.magicState = MAGIC_STATE_CONSUME_GORON_ZORA;
+            // fallthrough
+        case MAGIC_STATE_CONSUME_GORON_ZORA:
+            if ((play->pauseCtx.state == 0) && (play->pauseCtx.debugState == 0) && (msgCtx->msgMode == 0) &&
+                (play->gameOverCtx.state == 0) && (play->sceneLoadFlag == 0) && (play->unk_18B4A == 0)) {
+                if (!Play_InCsMode(play)) {
+                    interfaceCtx->magicConsumptionTimer--;
+                    if (interfaceCtx->magicConsumptionTimer == 0) {
+                        if (!(gSaveContext.save.weekEventReg[14] & 8)) {
+                            gSaveContext.save.playerData.magic--;
+                        }
+                        if (gSaveContext.save.playerData.magic <= 0) {
+                            gSaveContext.save.playerData.magic = 0;
+                        }
+                        interfaceCtx->magicConsumptionTimer = 10;
+                    }
+                }
+            }
+            if (!(gSaveContext.save.weekEventReg[14] & 8)) {
+                Magic_FlashMeterBorder();
+            }
+            break;
+
+        case MAGIC_STATE_CONSUME_GIANTS_MASK:
+            if ((play->pauseCtx.state == 0) && (play->pauseCtx.debugState == 0) && (msgCtx->msgMode == 0) &&
+                (play->gameOverCtx.state == 0) && (play->sceneLoadFlag == 0) && (play->unk_18B4A == 0)) {
+                if (!Play_InCsMode(play)) {
+                    interfaceCtx->magicConsumptionTimer--;
+                    if (interfaceCtx->magicConsumptionTimer == 0) {
+                        if (!(gSaveContext.save.weekEventReg[14] & 8)) {
+                            gSaveContext.save.playerData.magic--;
+                        }
+                        if (gSaveContext.save.playerData.magic <= 0) {
+                            gSaveContext.save.playerData.magic = 0;
+                        }
+                        interfaceCtx->magicConsumptionTimer = R_MAGIC_CONSUME_TIMER_GIANT;
+                    }
+                }
+            }
+            if (!(gSaveContext.save.weekEventReg[14] & 8)) {
+                Magic_FlashMeterBorder();
+            }
+            break;
+
+        default:
+            gSaveContext.magicState = MAGIC_STATE_IDLE;
+            break;
+    }
+}
+
+void Magic_DrawMeter(PlayState* play) {
+    InterfaceContext* interfaceCtx = &play->interfaceCtx;
+    s16 magicBarY;
+
+    OPEN_DISPS(play->state.gfxCtx);
+
+    if (gSaveContext.save.playerData.magicLevel != 0) {
+        if (gSaveContext.save.playerData.healthCapacity > 0xA0) {
+            magicBarY = 42; // two rows of hearts
+        } else {
+            magicBarY = 34; // one row of hearts
```

Comment:
Did these end up getting defines or not?

Reply:
In OoT, these were REG's which naturally got defines. In MM, they dropped the REGs here and just used hard constants

Reply:
They dropped a lot of REGs in parameter actually

Reply:
I think it wouldn't be a bad idea to have defines for all the numbers that were regs: it makes it easier to move code between, and it's clearer what they're for than having to comment on all of them (plus enables arithmetic if appropriate).